sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/HttpHelper.cs

```diff
@@ -275,5 +287,95 @@ internal static string GetUrlUsingHostAndPort(this AzMonList tagObjects, string
 
             return url;
         }
+
+        internal static string GetHostUsingNetPeerAttributes(this AzMonList tagObjects)
+        {
+            var netPeerName = AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeNetPeerName)?.ToString();
+            string host;
+            if (!string.IsNullOrWhiteSpace(netPeerName))
+            {
+                host = netPeerName;
+            }
+            else
+            {
+                host = AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeNetPeerIp)?.ToString();
+            }
+
+            return host;
+        }
+
+        ///<summary>
+        /// Gets http dependency target from activity tag objects.
+        ///</summary>
+        internal static string GetDependencyTarget(this AzMonList tagObjects, PartBType type)
+        {
+            string target;
+            string defaultPort;
+            switch (type)
+            {
+                case PartBType.Http:
+                    defaultPort = GetDefaultHttpPort(AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeHttpScheme)?.ToString());
+                    break;
+                case PartBType.Db:
+                    defaultPort = GetDefaultDbPort(AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeDbSystem)?.ToString());
+                    break;
+                default:
+                    defaultPort = "0";
+                    break;
+            }
+
+            var peerService = AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributePeerService)?.ToString();
+            if (!string.IsNullOrEmpty(peerService))
+            {
+                target = peerService;
+                return target;
+            }
+
+            if (type == PartBType.Http)
+            {
+                var httpHost = AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeHttpHost)?.ToString();
```

Comment:
Reviewed the source for complete file. Looks like we are reading `SemanticConventions.AttributeHttpHost` from the list at 2 different places for a single telemetry(`GetUrlUsingHttpHost` and `GetDependencyTarget`). Same for few other attributes too. Is it possible to refactor to read from the list once?

Reply:
That is a good point - will do some refactoring in separate PR.